redis/src/cluster_client.rs

```diff
@@ -1,164 +1,289 @@
+use std::time::Duration;
+
 use crate::cluster::ClusterConnection;
+use crate::connection::{ConnectionAddr, ConnectionInfo, IntoConnectionInfo};
+use crate::types::{ErrorKind, RedisError, RedisResult};
 
-use super::{
-    ConnectionAddr, ConnectionInfo, ErrorKind, IntoConnectionInfo, RedisError, RedisResult,
-};
+/// Redis cluster specific parameters.
+#[derive(Default, Clone)]
+pub(crate) struct ClusterParams {
+    pub(crate) password: Option<String>,
+    pub(crate) username: Option<String>,
+    pub(crate) read_from_replicas: bool,
+    pub(crate) tls_insecure: Option<bool>,
```

Comment:
I'm trying to figure out whether `tls_insecure` is actually an improvement to the API. It seems a bit less intuitive to me and I'm not sure what's gained by the change. :thinking:

Reply:
Currently, the user has to specify this flag in ConnectionAddr of ConnectionInfo. Instead of specifying it for each node, he can just specify the same flag here. We don't have to add this option to the builder, the main change was to simplify its use inside the connection impl.

Reply:
I would still argue that this should be left out for now, for the sake of minimizing the scope of changes in the PR.

Reply:
Removed it

Reply:
Thanks, will get my review finished up in the next couple days

Reply:
Sorry, maybe I wasn't clear in my previous comment. I was trying to say that I don't like the switch from using the `TlsMode` enum to `Option<bool>` and would prefer that the previous implementation be left in place. I think it would be great if we had a flag in the builder to indicate secure/insecure Tls, but would again, prefer that it use the previously defined enum. Perhaps I'm just focusing on a minor detail, but in my opinion the enum makes the code much clearer than `Option<bool`. @djc I'll defer to you here if you have any thoughts to the contrary. 

Other than that I think this PR looks great and would love to get it in.

Reply:
Done

Reply:
Done